src/components/Authentication/Login.jsx

```diff
@@ -0,0 +1,46 @@
+import { useLocation, useNavigate } from "react-router";
+import { loginHandler } from "../../utils";
+import {useLogin} from "../../context"
+const Login = () => {
+
+    const {loginState,loginDispatch} = useLogin();
+    const location = useLocation();
+    const navigate = useNavigate();
+
+    const showPasswordHandler = () => loginState.showPassword ? loginDispatch({showPassword: false}):loginDispatch({showPassword: true});
+    const setValueHandler = (e,field) => {
+        const fieldValue = {};
+        fieldValue[field] = e.target.value;
+        loginDispatch(fieldValue);
+    }
+    const setTestHandler = () => loginDispatch({email: "test@gmail.com", password: "test123"});
+    const setFocusHandler = (field, value) => {
+        const focusReset = { email: false, password: false}
+        focusReset[field] = value;
+        loginDispatch({focus: focusReset});
+    }    
+    
+    return (
+        <form onSubmit={(e) => loginHandler(e,location,navigate)} className="input-form login flex-column flex-gap-1 flex-wrap h-auto w-100">
+            <section className={`input-container flex-column m-5 ${loginState.email.length > 0 || loginState.focus.email ? "focused":"" }`}>
+                <input id="email" className="textbox-content p-5" type="email" name="email" onChange={(e) => setValueHandler(e,"email")} value={loginState.email} onFocus={() => setFocusHandler("email",true)} onBlur={()=> setFocusHandler("email",false)}/>
```

Comment:
I feel that these multiple handlers could be called separately. This would increase the readability and also keep your code cleaner.

Reply:
ok will look into this later